File: Yashwanth_24b1039/week2/mainak_array.py
from math import gcd
from math import lcm

n=int(input())

lst=list(map(int,input().split()))

gcd_prefix=[0]*n
gcd_suffix=[0]*n
gcd_prefix[0]=lst[0]
gcd_suffix[n-1]=lst[n-1]

for i in range(1,n):
    gcd_prefix[i]=gcd(gcd_prefix[i-1],lst[i])

for i in range(n-2,-1,-1):
    gcd_suffix[i]=gcd(gcd_suffix[i+1],lst[i])

gcd_remaining=[0]*n
gcd_remaining[0]=gcd_suffix[1]
result=gcd_remaining[0]
for i in range(1,n-1):
    gcd_remaining[i]=gcd(gcd_prefix[i-1],gcd_suffix[i+1])
    result=lcm(result,gcd_remaining[i])

result=lcm(result,gcd_prefix[n-2])

print(result)

File: Karan_24b3003/Week 2/mainak_array.cpp
#include <iostream>
#include <vector>
#include <numeric>
using namespace std;

long long gcd(long long a, long long b) {
    return b ? gcd(b, a % b) : a;
}
long long lcm(long long a, long long b) {
    return (a / gcd(a, b)) * b;
}
int main() {
    int n;
    cin >> n;
    vector<long long> a(n);

    for (int i = 0; i < n; ++i)
        cin >> a[i];
    long long g = a[0];
    for (int i = 1; i < n; ++i)
        g = gcd(g, a[i]);
    for (int i = 0; i < n; ++i)
        a[i] /= g;
    long long result = a[0];
    for (int i = 1; i < n; ++i) {
        long long l = lcm(a[0], a[i]);
        result = gcd(result, l);
    }
    cout << result * g << '\n';
    return 0;
}
